### PR TITLE
xcm query handler

### PR DESCRIPTION
**Pull request by muharem**

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2799613

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2802884

File: xcm/pallet-xcm/src/lib.rs [hunk 1]
```diff
@@ -52,8 +52,8 @@ use frame_system::pallet_prelude::*;
 pub use pallet::*;
 use xcm_executor::{
 	traits::{
-		CheckSuspension, ClaimAssets, DropAssets, FinishedQuery, MatchesFungible, OnResponse, QueryResponseStatus,
-		VersionChangeNotifier, WeightBounds, XcmQueryHandler,
+		CheckSuspension, ClaimAssets, DropAssets, MatchesFungible, OnResponse, QueryHandler,
+		QueryStatus as OuterQueryStatus, VersionChangeNotifier, WeightBounds,
```
Comment (tonyalaribe) [inline xcm/pallet-xcm/src/lib.rs:56]: Why did you alias it to OuterQueryStatus?

**Comment by tonyalaribe** [comment]
Looks good to me. Did you test it against the substrate and cumulus branches? If yes, then i would probably just merge this into the `cisco-xcm-query-handler` branch

File: xcm/pallet-xcm/src/lib.rs (hunk 1 above)
Comment (muharem) [reply xcm/pallet-xcm/src/lib.rs:56]: there is already a QueryStatus type inside the pallet. and I've seen in Frame we use Outer suffix (not exactly same case).

**Comment by muharem** [comment]
> Looks good to me. Did you test it against the substrate and cumulus branches? If yes, then i would probably just merge this into the `cisco-xcm-query-handler` branch

yes

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2812630

**Pull request merged**